src/flags.ml

```diff
@@ -3266,6 +3293,21 @@ let solver_dependant_actions () =
       )
     | None -> Log.log L_warn "Couldn't determine Boolector version"
   )
+  | `MathSAT_SMTLIB -> (
+    let cmd = Format.asprintf "%s -version" (Smt.mathsat_bin ()) in
+    match get_version true cmd with
+    | Some (major_rev, minor_rev, patch_rev) ->
+      if major_rev < 5 || (major_rev = 5 && (minor_rev < 5 || patch_rev < 4))  then (
```

Comment:
The right condition is:
`if major_rev < 5 || (major_rev = 5 && (minor_rev < 5 || (minor_rev = 5 && patch_rev < 4)))  then (`

Reply:
corrected. Thanks for pointing out.